src/libs/runtime-utils.js

```diff
@@ -77,14 +76,36 @@ export const getValidGpuTypes = (numCpus, mem, zone) => {
   return validGpuTypes || { name: '?', type: '?', numGpus: '?', maxNumCpus: '?', maxMem: '?', price: NaN, preemptiblePrice: NaN }
 }
 
-const gpuCost = (gpuType, numGpus) => _.find({ type: gpuType, numGpus }, gpuTypes)?.price || NaN
-
 const dataprocCost = (machineType, numInstances) => {
   const { cpu: cpuPrice } = findMachineType(machineType)
 
   return cpuPrice * numInstances * dataprocCpuPrice
 }
 
+const getHourlyCostForMachineType = (machineTypeName, region) => {
+  const machineType = _.find({ name: machineTypeName }, machineTypes)
+  const regionalPrices = _.find({ name: region }, regionToPrices)
+  return (machineType.cpu * regionalPrices.n1HourlyCpuPrice) + (machineType.memory * regionalPrices.n1HourlyGBRamPrice)
+}
+
+const getHourlyPreemptibleCostForMachineType = (machineTypeName, region) => {
+  const machineType = _.find({ name: machineTypeName }, machineTypes)
+  const regionalPrices = _.find({ name: region }, regionToPrices)
+  return (machineType.cpu * regionalPrices.preemptibleN1HourlyCpuPrice) + (machineType.memory * regionalPrices.preemptibleN1HourlyGBRamPrice)
+}
```

Comment:
`regionalPrices` `name`s are uppercase whereas values passed into the two methods above are lowercase which causes a runtime error. I find it difficult not to make mistakes when using various cases throughout the code requiring knowing where to make the right conversion at the right place; e.g. 

![image](https://user-images.githubusercontent.com/5438223/142275074-5bed5642-eeae-4a7c-8ee4-7ee34a773bcd.png)

Any reason why we can't default `name`s to lowercase and make conversions only at API call / display time if necessary?

Reply:
Also, does it make sense to merge these two helper methods as below to avoid code duplication?

```javascript
const getHourlyCostForMachineType = (machineTypeName, region, isPreemptible) => {
  const { cpu, memory } = _.find({ name: machineTypeName }, machineTypes)
  const { n1HourlyCpuPrice, preemptibleN1HourlyCpuPrice, n1HourlyGBRamPrice, preemptibleN1HourlyGBRamPrice } = _.find({ name: _.toUpper(region) },
    regionToPrices)
  return !isPreemptible ?
    (cpu * n1HourlyCpuPrice) + (memory * n1HourlyGBRamPrice) :
    (cpu * preemptibleN1HourlyCpuPrice) + (memory * preemptibleN1HourlyGBRamPrice)
}
```

Reply:
1) Where do you see lowercase values passed in? Names are always uppercase, and we convert to lowercase only for the leo api and for display time. We default to uppercase because the values returned by the Google API are uppercase.
2) Thanks! I have merged them together.

Reply:
> Where do you see lowercase values passed in?

While creating a Dataproc cluster.

Reply:
There's a bug with runtimeConfigCost. Its trying to compute cost based on the region returned by Leo's API. For dataproc clusters, the region returned is in lowercase. For standard VMs, the region doesn't exist (it returns zone instead), so it uses the default US-CENTRAL1 region (even if the VM is outside of US).

Working on a fix.

Reply:
Pushed a fix. Where we read the config from Leonardo directly, we turn the region (if exists) uppercase. If it doesn't exist (because its a standard VM), we get a region from the zone and uppercase that.